etc/deploy/aws.sh

```diff
@@ -71,20 +71,36 @@ parse_flags() {
 # $1 : bucket name (required)
 # $2 : boolean to use cloudfront or not (required)
 create_s3_bucket() {
-  if [[ "$#" -lt 1 ]]; then
+  if [[ "$#" -lt 1 ]] || [[ "${1}" =~ s3://* ]] ; then
     echo "Error: create_s3_bucket needs a bucket name"
     return 1
   fi
+  if [[ "$#" -lt 2 ]]; then
+    echo "Error: must specify whether cloudfront has access"
+    return 1
+  fi
+  if [[ "$#" -eq 3 ]]; then
+    KOPS_STATE_BUCKET="${3}"
+  fi
   BUCKET="${1#s3://}"
+  GIVE_CLOUDFRONT_ACCESS="${2}"
 
   # For some weird reason, s3 emits an error if you pass a location constraint when location is "us-east-1"
   if [[ "${AWS_REGION}" == "us-east-1" ]]; then
     aws s3api create-bucket --bucket ${BUCKET} --region ${AWS_REGION}
   else
     aws s3api create-bucket --bucket ${BUCKET} --region ${AWS_REGION} --create-bucket-configuration LocationConstraint=${AWS_REGION}
   fi
+  if [[ -n "${KOPS_STATE_BUCKET}" ]]; then
+    # Put bucket name in kops state store bucket right away, in case the cluster doesn't finish coming up
+    jq -n \
+      --arg pachyderm_bucket ${BUCKET} \
+      --arg timestamp "$(date)" \
+      '{"pachyderm_bucket": $pachyderm_bucket, "created": $timestamp}' \
+    | aws s3 cp - ${KOPS_STATE_BUCKET}/${NAME}-info.json
```

Comment:
This is outside of the scope of this PR, but I found the reliance on previously exported variables somewhat confusing

Reply:
Agreed. I added a TODO here (hopefully we rewrite this script in go someday soon anyway)